src/main/java/org/elasticsearch/index/query/QueryParsingException.java

```diff
@@ -19,21 +19,52 @@
 
 package org.elasticsearch.index.query;
 
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentLocation;
 import org.elasticsearch.index.Index;
 import org.elasticsearch.index.IndexException;
 import org.elasticsearch.rest.RestStatus;
 
+import java.io.IOException;
+
 /**
  *
  */
 public class QueryParsingException extends IndexException {
 
-    public QueryParsingException(Index index, String msg) {
-        super(index, msg);
+    public static final int UNKNOWN_POSITION = -1;
+    private int lineNumber = UNKNOWN_POSITION;
+    private int columnNumber = UNKNOWN_POSITION;
+
+    public QueryParsingException(Index index, String msg, @Nullable XContentLocation location) {
```

Comment:
+1 for tests, I am slightly worried that we will forget about this location parameter in some places in the future, was wondering if there's way to enforce it, so that we print out the location whenever possible.